operational/flexera/msp/customer_usage_insights_report/flexera_msp_customer_usage_insights_report.pt

```diff
@@ -0,0 +1,1425 @@
+    forbidden_org_ids = _.uniq(forbidden_org_ids);
+    var orgs_string = forbidden_org_ids.join(", ");
+    errors.push("403 Forbidden error received attempting to get user list from Flexera MSP Customer Orgs: "+orgs_string+". You should deploy this policy using a User Refresh Token credential associated with a User that has `org_owner` access in all MSP Child Orgs.  Alternatively, use the `Allow/Deny Child Orgs` parameter to filter the list of Child Orgs in scope.");
+  }
+
+  // Deduplicate identical errors
+  errors = _.uniq(errors);
+EOS
+end
+
+datasource "ds_children_org_get_roles_step1" do
+  iterate $ds_organizations_filtered
+  request do
+    auth $auth_flexera
+    host val($ds_flexera_api_hosts, 'flexera')
+    path join(["/iam/v1/orgs/", val(iter_item, "id"), "/roles"])
+    ignore_status 403 # Ignore status 403 so we can gracefully handle that kind of error
+  end
+  result do
+    encoding "json"
+    field "org", iter_item
+    field "roles", response
+  end
+end
+
+datasource "ds_children_org_get_billing_centers_step2" do
+  iterate $ds_children_org_get_roles_step1
+  request do
+    auth $auth_flexera
+    host rs_optima_host
+    path join(["/analytics/orgs/", val(val(iter_item, "org"), "id"), "/billing_centers"])
+    header "Api-Version", "1.0"
+    header "User-Agent", "RS Policies"
+    ignore_status [403]
+  end
+  result do
+    encoding "json"
+    field "org", val(iter_item, "org")
+    field "roles", val(iter_item, "roles")
+    field "billing_centers", response
+  end
+end
+
+# Gather top level billing center IDs for when we pull cost data
+datasource "ds_children_org_get_toplevel_billing_centers_step3" do
+  run_script $js_children_org_get_toplevel_billing_centers_step3, $ds_children_org_get_billing_centers_step2
+end
+
+script "js_children_org_get_toplevel_billing_centers_step3", type: "javascript" do
+  parameters "ds_children_org_get_billing_centers_step2"
+  result "result"
+  code <<-EOS
+  _.each(ds_children_org_get_billing_centers_step2, function(org) {
+    var top_level_bcs = _.filter(org['billing_centers'], function(bc) {
+      return bc['parent_id'] == null || bc['parent_id'] == undefined
+    })
+    org['top_level_bc_ids'] = _.pluck(top_level_bcs, 'id')
+  })
+  result = ds_children_org_get_billing_centers_step2
+EOS
+end
+
+datasource "ds_request_children_org_get_costs_step4" do
+  run_script $js_request_children_org_get_costs_step4, $ds_children_org_get_toplevel_billing_centers_step3, $ds_request_parent_org_costs_not_allocated, $param_get_costs_from_parent_org, rs_org_id, rs_org_name
+end
+
+script "js_request_children_org_get_costs_step4", type: "javascript" do
+  parameters "ds_children_org_get_toplevel_billing_centers_step3", "ds_request_parent_org_costs_not_allocated", "param_get_costs_from_parent_org", "rs_org_id", "rs_org_name"
+  result "result"
+  code <<-EOS
+  var result = [];
+
+  _.each(ds_children_org_get_toplevel_billing_centers_step3, function(org) {
```

Comment:
General feedback rather than repeating it. There is often a lack of empty lines in your code, and it can create an effect similar to reading one giant paragraph someone wrote when they didn't split their writing into separate paragraphs.

There should be an empty line after the end of code blocks and between sections of code to make it a bit more readable.